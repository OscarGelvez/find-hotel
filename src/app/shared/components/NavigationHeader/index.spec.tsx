import '@testing-library/jest-dom';

import { cleanup, getAllByTestId, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { NavigationHeader } from './index';

afterEach(cleanup);

const items = [
  { label: 'Habitaciones', url: '/rooms' },
  { label: 'Reservas', url: '/bookings' },
];

it('rederizar el componente NavigationHeader sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <NavigationHeader />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente NavigationHeader cuando se envían items', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <NavigationHeader />
    </BrowserRouter>
  );
  expect(getByTestId('div-header-nav')).toContainElement(
    getByTestId('div-brand')
  );

  expect(getByTestId('div-header-nav')).toContainElement(
    getByTestId('nav-list')
  );
});

it('Compara snapshot del componente NavigationHeader cuando se envían items', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <NavigationHeader />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
