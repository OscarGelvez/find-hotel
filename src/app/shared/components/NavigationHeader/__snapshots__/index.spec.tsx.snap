// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente NavigationHeader cuando se env√≠an items 1`] = `
<nav
  className="sc-Axmtr eubyYC"
  data-testid="div-header-nav"
>
  <div
    className="sc-AxirZ imQSvU"
    data-testid="div-brand"
  >
    <a
      className="sc-AxjAm jqjyPd"
      data-testid="link-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Find Hotel"
        className="sc-AxiKw iunsIC"
        data-testid="image-brand"
        src="logoHotel.png"
      />
    </a>
  </div>
  <ul
    className="sc-AxheI dkiJtT"
    data-testid="nav-list"
  >
    <li
      data-testid="nav-item-li"
    >
      <div
        className="sc-AxgMl cuEAew"
        data-testid="nav-item"
      >
        <a
          className="sc-AxjAm jqjyPd"
          data-testid="link-nav-item"
          href="/"
          onClick={[Function]}
        >
          Habitaciones
        </a>
      </div>
    </li>
    <li
      data-testid="nav-item-li"
    >
      <div
        className="sc-AxgMl cuEAew"
        data-testid="nav-item"
      >
        <a
          className="sc-AxjAm jqjyPd"
          data-testid="link-nav-item"
          href="/booking"
          onClick={[Function]}
        >
          Reservas
        </a>
      </div>
    </li>
  </ul>
</nav>
`;
